netbox/extras/signals.py

```diff
@@ -280,4 +292,39 @@ def process_job_end_event_rules(sender, **kwargs):
     """
     event_rules = EventRule.objects.filter(type_job_end=True, enabled=True, object_types=sender.object_type)
     username = sender.user.username if sender.user else None
-    process_event_rules(event_rules, sender.object_type.model, EVENT_JOB_END, sender.data, username)
+    process_event_rules(
+        event_rules=event_rules,
+        object_type=sender.object_type,
+        event_type=JOB_COMPLETED,
+        data=sender.data,
+        username=username
+    )
+
+
+#
+# Notifications
+#
+
+@receiver(post_save)
+def notify_object_changed(sender, instance, created, raw, **kwargs):
+    if created or raw:
+        return
+
+    # Skip unsupported object types
+    ct = ContentType.objects.get_for_model(instance)
+    if ct.model not in registry['model_features']['notifications'].get(ct.app_label, []):
+        return
+
+    # Find all subscribed Users
+    subscribed_users = Subscription.objects.filter(object_type=ct, object_id=instance.pk).values_list('user', flat=True)
+    if not subscribed_users:
+        return
+
+    # Delete any existing Notifications for the object
+    Notification.objects.filter(object_type=ct, object_id=instance.pk, user__in=subscribed_users).delete()
```

Comment:
So this would be one notification for an object regardless of what happens, so if I have different events one for create and one for update and I do both, I should only get a single notification?  We probably want to document that.

Reply:
I'm of the opinion that a user should generally see no more than one notification per object, but I think this depends somewhat on the event type. For example, you probably don't need to see separate notifications that an object was created _and_ later updated. But this assumption might not hold when we consider custom event types (being introduced in #16886).

I'm going to leave this as-is for now, with the expectation that we'll reassess the strategy once we get some real-world feedback during the beta evaluation.